autogen/main/firewall.tf.tmpl

```diff
@@ -48,9 +48,11 @@ resource "google_compute_firewall" "intra_egress" {
   allow { protocol = "esp" }
   allow { protocol = "ah" }
 
+  {% if not private_cluster %}
   depends_on = [
     google_container_cluster.primary,
   ]
+  {% endif %}
```

Comment:
Private clusters have a pre defined master cidr so no need to wait for cluster creation to determine master ip range.